common/libs/VkCodecUtils/VulkanVideoProcessor.cpp

```diff
@@ -770,3 +754,47 @@ VkResult VulkanVideoProcessor::ParseVideoStreamData(const uint8_t* pData, size_t
 
     return m_vkParser->ParseVideoData(&packet, pnVideoBytes, doPartialParsing);
 }
+
+void VulkanVideoProcessor::WaitForFrameCompletion(VulkanDecodedFrame* pFrame, VkSharedBaseObj<VkImageResource>& imageResource) {
+
+    const uint64_t fenceTimeout = 100 * 1000 * 1000; // 100 mSec
```

Comment:
We already have a function for that called WaitAndResetFence() Please use it.
I was about to add a function called WaitAndGetStatus() after, but forgot to do it.

Reply:
As you mention,WaitAndResetFence does not do the same thing. So we would need WaitAndGetStatus here.
This isn't really changing anything. We currently have this blob of code in the ConvertFrameToNv12 function. I was just attempting to make the function smaller, and this can go away when you add WaitAndGetStatus().

I'll add a TODO:

Reply:
You can use WaitAndResetFence() for the first part, or add the WaitAndGetStatus() utility function.